### PR TITLE
Update Toolbar component readme

### PR DESCRIPTION
**Pull request by kjellr**
Adding documentation to describe the use and functionality of the Toolbar component.

Thanks @cburton4 and @alexislloyd for drafting this.

### CODE REVIEW

**Review by melchoyce — APPROVED** [review]
Everything looks good to me 👍

Edit: Actually, seeing heading issues in here.

File: packages/components/src/toolbar/README.md [hunk 1]
```diff
@@ -1,5 +1,44 @@
...
+Toolbar can be used to group related options. To emphasize groups of related icon buttons, a toolbar should share a common container.
+
+![Toolbar component above an image block](https://wordpress.org/gutenberg/files/2019/01/s_96EC471FE9C9D91A996770229947AAB54A03351BDE98F444FD3C1BF0CED365EA_1541782974545_ButtonGroup.png)
+
+## Table of contents
+
+1. [Design guidelines](#design-guidelines)
+2. [Development guidelines](#development-guidelines)
+
+# Design guidelines
```
Comment (melchoyce) [inline packages/components/src/toolbar/README.md]: Heading level issue

**Review by melchoyce — CHANGES_REQUESTED** [review]

File: packages/components/src/toolbar/README.md (hunk 1 above)
Comment (kjellr) [reply packages/components/src/toolbar/README.md]: Good catch! Just pushed b1e320c to fix that. Might giving it one more review? Thank you!

File: packages/components/src/toolbar/README.md [hunk 2]
```diff
@@ -1,6 +1,45 @@
...
+
+- **Clearly communicate that clicking or tapping will trigger an action.**
+- **Use established colors appropriately.** For example, only use red for actions that are difficult or impossible to undo.
+- When used with a block, have a consistent location above the block. **Otherwise, have a consistent location in the interface.**
+
+![A toolbar attached to the top left side of a block. (1. Toolbar, 2. Block)](https://wordpress.org/gutenberg/files/2019/01/s_D8D19E5A314C2D056B8CCC92B2DB5E27164936A0C5ED98A4C2DFDA650BE2A771_1542388042335_toolbar-block.png)
+
+### States
+
+**Active and available toolbars**
```
Comment (chrisvanpatten) [inline packages/components/src/toolbar/README.md]: Should these be h4s now similar to #13334?

Comment (kjellr) [reply packages/components/src/toolbar/README.md]: Yes, they should! Updated in b1e320c. Thanks

**Review by chrisvanpatten — APPROVED** [review]
Looks great!

**Review by melchoyce — APPROVED** [review]
LGTM!

**Comment by kjellr** [comment]
Thank you both!

**Pull request merged**